src/rust/ide/lib/ast/impl/src/lib.rs

```diff
@@ -1187,7 +1187,12 @@ impl Block<Ast> {
         let empty_lines = Vec::new();
         let first_line  = BlockLine::new(first_line.clone_ref());
         let lines       = tail_lines.iter().cloned().map(BlockLine::new).collect();
-        let is_orphan   = tail_lines.is_empty();
+        // We virtually never want block to be an orphan (i.e. start with a first line attached
+        // to whatever AST was there before). While it may seem tempting to have this for single
```

Comment:
I don't understand this "start with a first line attached to whatever AST was there before" I see the comment that orphan is "Block start with a leading newline" what does not make sense to me as well, especially comparing with the function docs.

Reply:
The documentation for the block is incorrect. The leading newline is present when the `is_orphan` is false, not the other way around.
As for clarifying this comment, let's discuss this tomorrow.